oauth2orize/index.d.ts

```diff
@@ -0,0 +1,93 @@
+// Type definitions for oauth2orize v1.5.1
+// Project: https://github.com/jaredhanson/oauth2orize/
+// Definitions by: Wonshik Kim <https://github.com/wokim/>, Kei Son <https://github.com/heycalmdown>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as http from "http";
+
+declare module "oauth2orize" {
+
+  interface ServerOptions {
+    store: any;
+    loadTransaction: boolean;
+  }
+  export function createServer(options?: ServerOptions): OAuth2Server;
+
+  export interface AuthorizeOptions {
+    idLength?: number;
+    sessionKey?: string;
+  }
+
+  export interface ErrorHandlerOptions {
+    mode?: string;
+  }
+
+  export class OAuth2Server {
+    exchange(fn: ExchangeFunction): OAuth2Server;
+    exchange(type: string, fn: ExchangeFunction): OAuth2Server;
+    // Parses requests to obtain authorization
+    authorize(options: AuthorizeOptions, validate: (clientId: string, redirectURI: string, validated: (err: Error, client: any, redirectURI: string) => void) => void): Function;
+    authorize(validate: (clientId: string, redirectURI: string, validated: (err: Error | null, client?: any, redirectURI?: string) => void) => void): Function;
+    //authorization(options: AuthorizeOptions, validate: (clientId: string, redirectURI: string, validated: (err: Error, client: any, redirectURI: string) => void) => void): Function;
+    //authorization(validate: (clientId: string, redirectURI: string, validated: (err: Error, client: any, redirectURI: string) => void) => void): Function;
+    token(options?: any): (req: http.IncomingMessage, res: http.ServerResponse, next: any) => void;
+    errorHandler(options?: any): (err: Error, req: http.IncomingMessage, res: http.ServerResponse, next: any) => void;
+    serializeClient(fn: (client: any, done: (err: Error | null, id: string) => void) => void): void;
+    serializeClient(client: any, done: (err: Error | null, id: string) => void): void;
+    deserializeClient(fn: (id: string, done: (err: Error | null, client?: any | boolean) => void) => void): void;
+    deserializeClient(obj: any, done: (err: Error | null, client?: any | boolean) => void): void;
+  }
+
+  interface ExchangeFunction extends Function {
```

Comment:
Why not just use `type ExchangeFunction = (req: http.ServerRequest, res: http.ServerResponse, next: Function) => void;`?
Also, is there a more specific type `next` could have?